### ISSUE TITLE
Classical ML

### ISSUE THREAD

**Issue by mjovanc** [blackportal-ai/delta#221]
Features

- [ ] Linear Regression
- [ ] Logistic Regression
- [ ] Decision Trees
- [ ] Random Forest
- [ ] Support Vector Machines (SVM)
- [ ] K-Nearest Neighbors (KNN)
- [ ] Naive Bayes
- [ ] K-Means Clustering
- [ ] Principal Component Analysis (PCA)
- [ ] Gradient Boosting (e.g., XGBoost, LightGBM)

**Comment by sutt0n** [comment]
i've added a PR for random forest here: https://github.com/blackportal-ai/delta/pull/241

**Comment by sutt0n** [comment]
if you'd like, you can assign this to me

**Comment by Demon-Sheriff** [comment]
@mjovanc @chaseWillden  added a PR for Naive Bayes, check it here: #255
